packages/docusaurus/src/webpack/plugins/CleanWebpackPlugin.ts

```diff
@@ -115,6 +116,19 @@ export default class CleanWebpackPlugin {
     this.removeFiles = this.removeFiles.bind(this);
   }
 
+  private checkIfFileNameIsDuplicate(): boolean {
+    const dir = path.dirname(this.outputPath);
+    const baseName = path.basename(this.outputPath).toLowerCase();
+    // eslint-disable-next-line no-restricted-properties
+    const duplicateFiles = fs
+      .readdirSync(dir)
+      .filter((fileName) => fileName.toLowerCase() === baseName);
+    if (duplicateFiles.length > 0) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
I think this should be enough

```tsx
if (
      (await fs.pathExists(outputPath)) &&
      (await fs.lstat(outputPath)).isFile()
    ) {
   throw new Error("...");
}
```

On:
- case insensitive FS: this should throw for files like `BUILD` or `build`
- case-sensitive FS: if a file named `BUILD` exists, I think it's allowed to create a folder named `build`? 🤔 


I only have access to a case insensitive file-system. Can someone test my assumption on case sensitive file-system?

Reply:
@slorber, I think yours is better because it doesn't break the abstraction of whether the FS is case-sensitive or not :)

Reply:
> I think this should be enough
> 
> ```tsx
> if (
>       (await fs.pathExists(outputPath)) &&
>       (await fs.stat(outputPath)).isFile()
>     ) {
>    throw new Error("...");
> }
> ```
> 
> On:
> 
> * case insensitive FS: this should throw for files like `BUILD` or `build`
> * case-sensitive FS: if a file named `BUILD` exists, I think it's allowed to create a folder named `build`? 🤔
> 
> I only have access to a case insensitive file-system. Can someone test my assumption on case sensitive file-system?

I used [gitpod.io](gitpod.io) to test this usecase. It is a case-sensitive system

Reply:
What do you think about this, any suggestions? @slorber @Josh-Cena

Reply:
@Josh-Cena I see, keeping it simple then!

Reply:
@Josh-Cena @slorber I have used the sync version of `pathExists` and `stat` functions and fixed the error message. Let me know if this looks good.

Reply:
I have also tested the code in my local env. Works fine:)